ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c

```diff
@@ -101,44 +109,43 @@ VL53L0X_Error VL53L0X_UnlockSequenceAccess(VL53L0X_DEV Dev){
     return Status;
 }
 
-// the ranging_sensor_comms.dll will take care of the page selection
 VL53L0X_Error VL53L0X_WriteMulti(VL53L0X_DEV Dev, uint8_t index, uint8_t *pdata, uint32_t count){
 
     VL53L0X_Error Status = VL53L0X_ERROR_NONE;
     int32_t status_int = 0;
-	uint8_t deviceAddress;
+    uint8_t I2CBuffer[count+1];
 
     if (count>=VL53L0X_MAX_I2C_XFER_SIZE){
         Status = VL53L0X_ERROR_INVALID_PARAMS;
     }
 
-	deviceAddress = Dev->I2cDevAddr;
+    I2CBuffer[0] = index;
+    memcpy(&I2CBuffer[1], pdata, count);
 
-	status_int = VL53L0X_write_multi(deviceAddress, index, pdata, count);
+    status_int = i2c_write(Dev->i2c, I2CBuffer, count+1, Dev->I2cDevAddr);
 
-	if (status_int != 0)
-		Status = VL53L0X_ERROR_CONTROL_INTERFACE;
+    if (status_int < 0){
+        Status = VL53L0X_ERROR_CONTROL_INTERFACE;
+         SYS_LOG_ERR("Failed to write");
+    }
 
     return Status;

```

Comment:
why modify all this, why not just provide a version of VL53L0X_write_multi?

Reply:
The adaptation of the lib to Zephyr has to be done in ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c. That's why I call Zephyr function from here.
The usage of VL53L0X_write_multi is not a good option for me. It is part of vl53l0x_i2c_win_serial_comms.c file, which is present in ST delivery but this is just presented as an example. When reading the doc, they explain that we need to do the adaptation in vl53l0x_platform.c.

Regarding this, are you OK to keep my implementation ?

Reply:
if you feel like you need a full implementation of vl53l0x_platform.c than do it in the zephyr driver, not here.

Reply:
Done in the last version. I have moved this file to driver and cleaned it.